csharp_dotnetcore/11.QnAMaker/QnAMaker/QnABot.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+
+namespace AspNetCore_QnA_Bot
+{
+    /// <summary>
+    /// Represents a bot that can process incoming activities.
+    /// For each interaction from the user, an instance of this class is called.
```

Comment:
I might consider adding more detail here describing the object lifecycle.
For each Activity received, a new instance of this class is created. Objects that are expensive to construct, or have a lifetime beyond the single Turn, should be carefully managed.